staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1406,14 +1419,66 @@ func CreateThreeWayMergePatch(original, modified, current []byte, dataStruct int
 	return json.Marshal(patchMap)
 }
 
-func ItemAddedToModifiedSlice(original, modified string) bool { return original > modified }
+func ItemAddedToModifiedSlice(compareResult mergeKeysCompareResultType) bool {
+	return compareResult == mergeKeysGreater
+}
+
+func ItemRemovedFromModifiedSlice(compareResult mergeKeysCompareResultType) bool {
+	return compareResult == mergeKeysLess
+}
+
+func ItemMatchesOriginalAndModifiedSlice(compareResult mergeKeysCompareResultType) bool {
+	return compareResult == mergeKeysEqual
+}
+
+type mergeKeysCompareResultType string
+
+const (
+	mergeKeysLess     mergeKeysCompareResultType = "mergeKeysLess"
+	mergeKeysGreater  mergeKeysCompareResultType = "mergeKeysGreater"
+	mergeKeysEqual    mergeKeysCompareResultType = "mergeKeysEqual"
+	mergeKeysNotFound mergeKeysCompareResultType = "mergeKeysNotFound"
+)
 
-func ItemRemovedFromModifiedSlice(original, modified string) bool { return original < modified }
+func compareByMergeKeys(left, right map[string]interface{}, mergeKeys []string) mergeKeysCompareResultType {
+	var leftValue, rightValue interface{}
+	var leftString, rightString string
+	var foundLeft, foundRight bool
+	for _, mergeKey := range mergeKeys {
+		if leftValue, foundLeft = left[mergeKey]; foundLeft {
+			leftString = fmt.Sprintf("%v", leftValue)
+		} else {
+			return mergeKeysNotFound
+		}
+		if rightValue, foundRight = right[mergeKey]; foundRight {
+			rightString = fmt.Sprintf("%v", rightValue)
+		} else {
+			return mergeKeysNotFound
+		}
+		if leftString != rightString {
+			if sort.StringsAreSorted([]string{leftString, rightString}) {
+				return mergeKeysLess
+			} else {
+				return mergeKeysGreater
+			}
+		}
+	}
+	return mergeKeysEqual
+}
 
-func ItemMatchesOriginalAndModifiedSlice(original, modified string) bool { return original == modified }
+func CreateDeleteDirective(mergeKeys []string, mergeKeysMap map[string]interface{}) map[string]interface{} {
+	m := map[string]interface{}{directiveMarker: deleteDirective}
+	addMergeKeysToMap(m, mergeKeys, mergeKeysMap)
+	return m
+}
 
-func CreateDeleteDirective(mergeKey string, mergeKeyValue interface{}) map[string]interface{} {
-	return map[string]interface{}{mergeKey: mergeKeyValue, directiveMarker: deleteDirective}
+func addMergeKeysToMap(m map[string]interface{}, mergeKeys []string, mergeKeysMap map[string]interface{}) {
+	for _, mergeKey := range mergeKeys {
+		mergeKeyValue, found := mergeKeysMap[mergeKey]
+		if found {
+			m[mergeKey] = mergeKeyValue
+		}
```

Comment:
It is not allowed in https://github.com/kubernetes/community/blob/master/contributors/design-proposals/multi-fields-merge-key.md.
I will update the code.

But we may allow that as described in https://github.com/kubernetes/community/pull/476